int16 id
int16 pin
int16 value